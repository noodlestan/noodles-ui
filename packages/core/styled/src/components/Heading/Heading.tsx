import {
    type HeadingLevel,
    Heading as UnstyledHeading,
    type HeadingProps as UnstyledHeadingProps,
} from '@noodles-ui/core-unstyled';
import { Component } from 'solid-js';

import { contentColorClassList } from '../../variants';

import styles from './Heading.module.scss';

export type HeadingProps = UnstyledHeadingProps & {
    variant: string;
    color: string;
    level?: HeadingLevel;
};

const MAP_VARIANT_TO_LEVEL: Record<string, HeadingLevel> = {
    page: 1,
    section: 2,
    group: 3,
    item: 4,
};

export const Heading: Component<HeadingProps> = props => {
    const level = () => props.level || MAP_VARIANT_TO_LEVEL[props.variant];

    const classList = () => ({
        [styles.Heading]: true,
        [styles[`Heading-variant-${props.variant}`]]: true,
        ...contentColorClassList(props.color),
    });
    return <UnstyledHeading classList={classList()} {...props} level={level()} />;
};
