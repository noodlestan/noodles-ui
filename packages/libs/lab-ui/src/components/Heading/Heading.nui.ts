import { HeadingResource as HeadingStyledResource } from '@noodles-ui/core-styled';
import { ComponentExtendResource } from '@noodles-ui/core-types';

export const HeadingResource: ComponentExtendResource = {
    module: '@noodles-ui/lab-ui',
    extend: HeadingStyledResource,
    hides: {
        classList: {},
        style: {},
    },
    overrides: {
        variant: {
            name: 'HeadingVariant',
            options: ['large', 'small'],
            vars: { family: 'heading' },
            defaultValue: 'large',
        },
        tag: {
            name: 'HeadingTag',
            options: ['h1', 'h2', 'h3'],
        },
    },
};
