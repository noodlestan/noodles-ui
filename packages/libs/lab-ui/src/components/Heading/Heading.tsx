/**
 * noodles-ui / auto-generated
 *
 * 2024-03-20T23:32:09.835Z
 * <project>/src/components/Heading/Heading.tsx
 */

export { Heading, HeadingProps } from '../../generated/component.Heading';
