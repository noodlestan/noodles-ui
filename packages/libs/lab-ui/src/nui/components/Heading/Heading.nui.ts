import { HeadingResource as HeadingStyledResource } from '@noodles-ui/core-styled';
import { ComponentExtendResource } from '@noodles-ui/core-types';

export const HeadingResource: ComponentExtendResource = {
    module: '@noodles-ui/lab-ui',
    extend: HeadingStyledResource,
    hides: {
        classList: {},
        style: {},
    },
    overrides: {
        variant: {
            name: 'HeadingVariant',
            options: ['page', 'section', 'group', 'item'],
            defaultValue: 'page',
        },
        tag: {
            name: 'HeadingTag',
            options: ['h1', 'h2', 'h3'],
        },
    },
    vars: {
        family: 'heading',
    },
};
