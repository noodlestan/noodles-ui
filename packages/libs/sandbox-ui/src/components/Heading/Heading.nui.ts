import { HeadingResource as HeadingStyledResource } from '@noodles-ui/core-styled';
import { ComponentResource, VariantInlineResource } from '@noodles-ui/core-types';

const variant: VariantInlineResource = {
    type: 'variant',
    options: ['large', 'small'],
    defaultOption: 'large',
};

export const HeadingResource: ComponentResource = {
    module: '@noodles-ui/sandbox-ui',
    extend: HeadingStyledResource,
    overrides: {
        variant,
    },
};
