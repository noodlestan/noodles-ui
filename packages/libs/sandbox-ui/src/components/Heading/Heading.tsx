/**
 * noodles-ui / auto-generated
 *
 * 2024-03-11T14:36:34.627Z
 * <project>/src/components/Heading/Heading.tsx
 */

export { Heading } from '../../generated/component.Heading';
