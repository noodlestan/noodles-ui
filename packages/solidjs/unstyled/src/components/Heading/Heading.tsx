import { Component, JSX } from 'solid-js';
import { Dynamic } from 'solid-js/web';

export type HeadingLevel = '1' | '2' | '3' | '4';

export type HeadingProps = {
    level: HeadingLevel;
    tag?: string;
    children?: JSX.Element;
    classList?: { [key: string]: boolean };
    style?: JSX.CSSProperties;
};

const MAP_LEVEL_TO_TAG: Record<HeadingLevel, string> = {
    1: 'h1',
    2: 'h2',
    3: 'h3',
    4: 'h4',
};

export const Heading: Component<HeadingProps> = props => {
    const tag = () => props.tag || MAP_LEVEL_TO_TAG[props.level];

    return (
        <Dynamic component={tag()} classList={props.classList} style={props.style}>
            {props.children}
        </Dynamic>
    );
};
