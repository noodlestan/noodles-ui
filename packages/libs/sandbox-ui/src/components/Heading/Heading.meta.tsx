import { HeadingMeta as HeadingStyledMeta } from '@noodles-ui/core-styled';

import { Heading as Component } from '../../generated/component.Heading';

export const HeadingMeta = {
    type: 'component',
    name: 'Heading',
    extend: HeadingStyledMeta,
    component: Component,
    props: {
        variant: {
            options: ['large', 'small'],
            defaultOption: 'large',
        },
    },
};
