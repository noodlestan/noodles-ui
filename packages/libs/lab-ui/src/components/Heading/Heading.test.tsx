import { render } from '@solidjs/testing-library';
import { describe, expect, test } from 'vitest';

import { Heading } from './Heading';

const content = 'Lorem ipsum';

describe('<Heading />', () => {
    test('renders', () => {
        const { container, unmount } = render(() => (
            <Heading level="1" variant="page" tag="h2">
                {content}
            </Heading>
        ));
        expect(container.innerHTML).toMatchSnapshot();
        unmount();
    });
});
