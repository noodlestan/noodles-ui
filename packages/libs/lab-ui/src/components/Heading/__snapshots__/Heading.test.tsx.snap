// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Heading /> > renders 1`] = `"<h2 class="_Heading_ad18fc _Heading-variant-large_ad18fc">Lorem ipsum</h2>"`;
